src/Configuration/Properties/Property.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+declare(strict_types=1);
+
+namespace MyParcelCom\Integration\Configuration\Properties;
+
+class Property
+{
+    public function __construct(
+        public readonly string $name,
+        public readonly PropertyType $type,
+        public readonly string $description,
+        public readonly bool $isRequired = false,
+        public readonly bool $isPassword = false,
+        public readonly array $enum = [],
+        public readonly string $hint = '',
+    ) {
+    }
+
+    public function toArray(): array
+    {
+        return [
+            $this->name => array_filter([
+                'type' => $this->type->value,
+                'description' => $this->description,
+                'enum' => $this->enum,
+                'meta' => array_filter([
+                    'hint' => $this->hint,
+                    'password' => $this->isPassword,
+                ]),
+            ])
+        ];
+    }
+}
```

Comment:
Having the parent class responsible for the serialization of all fields makes it aware of all possible subtypes.
For example, password is only a case for password fields. Somehow now the parent Property class knows about it. Same goes for the enum. That somewhat dismisses the benefits of having subtypes like Checkbox or Number since what defines a checkbox is now both in the Checkbox class, and somewhat in the parent Property.

How about the following idea: each field class has it's own `toJsonSchema()` or `jsonSchemaProperty()` function where the serialization happens exactly for that case. My motivation for that comes from the fact that if we proceed to enrich the parent field class with more knowledge of possible configurations at some point it will know all the ins-and-outs of every subtype, dismissing the subtype idea little by little.

Now, lets imagine that the refactor I describe above gets done, the result creates two further disadvantages:
- these serialization methods have some small, but significant duplication of code in each subtype - fields `type`, `description` for example are always required therefore each subtype has to make them;
- if you leave the serialization methods to return an array there is nothing enforcing a structure that has the minimum required fields for a valid json schema (such as `type` and `description`). 

From there a follow-up refactor will probably show up, introducing a json-schema tight _Property_ class that can be returned by the serialization methods. 

After all this evolution through refactoring I think a very important realization bubbles up to the surface. That is that we have two concepts that together make this entire structure:
- The user-developer facing interface for form creation
- The transportation/interpretation through json-schema

The code can reflect these concepts since they have definable boundaries. Therefore the concepts above for me translate to class structure that reflects them directly:
- Configuration, Fields - what the user-developer utilizes and works with
- JsonSchema, Property - what the configuration and fields compile into

Since we reached a point where both of these concepts are now much more visible, at least from my point of view, their code analogs solidify (the implementation conveys more confidence).